Deployment/ASP.Net_Core_to_EC2/Readme.md

```diff
@@ -3,215 +3,436 @@
+
+    ![codedeploy_search](./media/codedeploy.png)
+
+1.  Go to Applications, click Create application
+
+    ![get_started_now](./media/cd_app.png)
+
+1.  Enter the Application name
+1.  Select EC2/On-premises under Compute platform and select Create application
+
+    ![sample_deployment](./media/cd_create_app.png)
+
+1.  Go to the application and select Create deployment group
+
+    ![inplace_deployment](./media/cd_create_grp.png)
+
+1.  Enter the Deployment group name
+1.  Select the Service Role created in Module 7
+
+    ![windows_type](./media/cd_grp1.png)
+
+1.  Choose In-place deployment.
+1.  Select Amazon EC2 instances under Environment configuration
+1.  Add a tag with Key - Name, Value - (Name of the EC2 instance created in Module 6) This will identify the EC2 instance created in Module 6
+
+    ![select_windows_sample](./media/cd_grp2.png)
+
+1.  Select Never under Agent configuration with AWS Systems Manager as we have already configured the EC2 instance with the CodeDeploy Agent.
+1.  Select CodeDeployDefault.AllAtOnce under Deployment Settings.
+1.  Deselect Enable load balancing.
+1.  Select Create deployment group
+
+    ![service_role](./media/cdgrp3.png)
+
+1.  CodeDeploy is ready to receive our application.
+
+### Module 9: Create an Amazon S3 bucket
+
+#### Overview
+
+In this module, we will create an Amazon S3 bucket to which the revision bundle will be uploaded. A revision in AWS CodeDeploy is a version of an application you want to deploy. Our sample application revisions are stored in Amazon S3
+
+#### Implementation
+
+1. From the AWS Console search for S3.
+1. Select Create bucket.
+
+    ![s3_bucket](./media/s31.png)
+
+1. Enter the Bucket name and select the AWS Region
+
+    ![s3_bucket](./media/s32.png)
+
+1. Leave the remaining settings to default and click Create bucket
+
+    ![s3_bucket](./media/s33.png)
+
+### Module 10: Integrate Azure DevOps and AWS CodeDeploy
+
+#### Overview
+
+In this step we will create an Azure DevOps build job that will deploy our
 application to EC2 instances via CodeDeploy.
 
 #### Implementation Instructions
 
-1.  Open our repo in VSTS, click on the Builds and Releases tab. Click
-    New.
+1.  Open our project in Azure DevOps, click on Pipelines. Click Create Pipeline.
+
+    ![new_build](./media/pipeline1.png)
+
+1.  Select Other Git.
 
-    ![new_build](./media/image19.png)
+    ![vsts_git](./media/pipeline2.png)
 
-1.  VSTS Git is our Source repo and we want to publish the master
-    branch. Click Continue.
+1.  Choose the Azure Repos Git, select your project and Repository
+1.  Select Continue
 
-    ![vsts_git](./media/image20.png)
+    ![.net_template](./media/pipeline3.png)
 
 1.  Choose the ASP<span></span>.NET template and click Apply.
 
-    ![.net_template](./media/image21.png)
+    ![add_task](./media/image21.png)
+
+
+1. Go to the task Build solution and append the following in MSBuild Arguments. This will set the default path where our application will be hosted
```

Comment:
It would be really helpful to users to point out the key settings they should use, in conjunction with the screenshot. Also, did we resort to putting the app output into $(Build.ArtifactStagingDirectory), or a subfolder of that location?

Reply:
Yes we did try both the ways - with the app and without it. Both work!